rasa/nlu/training_data/converters/watson_nlu_json_to_yaml_converter.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import json
+import os
+from typing import Any, Dict, List, Text
+from pathlib import Path
+from rasa.shared.nlu.constants import INTENT, ENTITIES, TEXT
+from rasa.shared.nlu.training_data.util import transform_entity_synonyms
+from rasa.shared.nlu.training_data.training_data import TrainingData
+from rasa.shared.nlu.training_data.message import Message
+from rasa.utils.converter import TrainingDataConverter
+from rasa.shared.nlu.training_data.formats.rasa_yaml import RasaYAMLWriter
+
+logger = logging.getLogger(__name__)
+
+
+class WatsonTrainingDataConverter(TrainingDataConverter):
+    """Reads Watson training data and train a Rasa NLU model."""
+
+    def filter(self, source_path: Path) -> bool:
+        """Checks if the given training data file Watson NLU Data.
+
+        Args:
+            source_path: Path to the training data file.
+
+        Returns:
+            `True` if the given file can be converted, `False` otherwise
+        """
+
+        if os.path.isfile(source_path):
```

Comment:
As we are already having a `Path` object we can simplify the code a bit
```suggestion
        if source_path.is_file:
```